ethcoder/abi_helpers.go

```diff
@@ -459,7 +459,7 @@ func ABIUnmarshalStringValuesAny(argTypes []string, stringValues []any) ([]any,
 			switch v := v.(type) {
 			case string:
 				// v is string array, ie. `["1","2","3"]`
-				err = json.Unmarshal([]byte(v), &stringValues)
+				err = sonic.ConfigDefault.Unmarshal([]byte(v), &stringValues)
```

Comment:
lets use sonic.ConfigFastest

Reply:
We can't do use the fastest because of the `NoQuoteTextMarshaler` option on.
With the option on, it does not check if a `MarshalText()` result needs quote.
We can either have this option off, or add the quotes to each of methods.

Reply:
Can you send an example MarshalText() method that causes the panic, please?

Reply:
You can try out this snipped to see an example.

```go
package main

import (
	"fmt"

	"github.com/bytedance/sonic"
)

type A string

func (a A) MarshalText() (text []byte, err error) {
	return []byte(a), nil
}

type B string

func (b B) MarshalText() (text []byte, err error) {
	buf := make([]byte, 0, len(b)+2)
	buf = append(buf, '"')
	buf = append(buf, b...)
	buf = append(buf, '"')
	return buf, nil
}

func main() {
	raw, err := sonic.ConfigFastest.Marshal(A("banana"))
	if err != nil {
		panic(err)
	}
	fmt.Printf("%s\n", raw)

	raw, err = sonic.ConfigFastest.Marshal(B("banana"))
	if err != nil {
		panic(err)
	}
	fmt.Printf("%s\n", raw)
}
```
The current behavior is the one in `A` in our types, the fast configuration expects B’s behavior.

Reply:
https://golang.bg/src/encoding/json/encode.go?s=9213%3A9305

```
// If no [Marshaler.MarshalJSON] method is present but the
// value implements [encoding.TextMarshaler] instead, Marshal calls
// [encoding.TextMarshaler.MarshalText] and encodes the result as a JSON string.
```

This means that `MarshalText()` should not return `"string"` but it should return `string` and the encoder will encode it as JSON string.

If we ever need to marshal some other type (object/number/array), we'd have to use MarshalJSON() instead.


TL;DR: Let's fix our `TextMarshallers()`.

Reply:
Note that this was changed recently upstream: https://github.com/bytedance/sonic/pull/781

So please update to latest Sonic :)